lib/Sema/BuilderTransform.cpp

```diff
@@ -918,13 +918,11 @@ std::optional<BraceStmt *>
 TypeChecker::applyResultBuilderBodyTransform(FuncDecl *func, Type builderType) {
   // First look for any return statements, and bail if we have any.
   auto &ctx = func->getASTContext();
-  if (evaluateOrDefault(ctx.evaluator, BraceHasReturnRequest{func->getBody()},
-                        false)) {
+  if (SmallVector<ReturnStmt *> returnStmts;
+      func->getBody()->getExplicitReturnStmts(ctx, returnStmts),
+      !returnStmts.empty()) {
```

Comment:
Cute, but I would rather just bind `returnStmts` outside the `if`, up to you though